lib/codegen.ex

```diff
@@ -48,14 +48,15 @@ defmodule Jason.Codegen do
       |> Enum.map(&encode_pair(&1, encode_args))
       |> Enum.intersperse(",")
 
-    collapse_static(List.flatten(["{", elements] ++ '}'))
+    collapse_static(List.flatten(["{", elements] ++ ~c'}'))
   end
 
-  defp clauses_to_ranges([{:->, _, [[{:in, _, [byte, range]}, rest], action]} | tail], acc) do
-    clauses_to_ranges(tail, [{range, {byte, rest, action}} | acc])
+  defp clauses_to_ranges([{:->, _, [[{:in, _, [byte, range]}, rest], action]} | tail], acc, env) do
+    range = Macro.expand(range, env)
```

Comment:
Unlike charlists, sigils are not their own AST, so we need to expand them first.